src/python_testing/TC_TSTAT_4_2.py

```diff
@@ -647,6 +642,28 @@ async def test_TC_TSTAT_4_2(self):
         self.step("18")
         if self.pics_guard(self.check_pics("TSTAT.S.F08") and self.check_pics("TSTAT.S.A0050") and self.check_pics("TSTAT.S.Cfe.Rsp")):
 
+            # Generate list of tuples of scenarios and number of remaining presets per scenario allowed
+            presetScenarioHeadroom = list((presetType.presetScenario,
+                                           presetType.numberOfPresets - presetScenarioCounts.get(presetType.presetScenario, 0)) for presetType in presetTypes)
+
+            if len(presetScenarioHeadroom) > 0:
+                # Find the preset scenario with the smallest number of remaining allowed presets
+                presetScenarioHeadroom = sorted(presetScenarioHeadroom, key=lambda diff: diff[1])
+                (presetScenario, headroom) = presetScenarioHeadroom[0]
+
+                # Add one more preset than is allowed by the preset type
+                test_presets = copy.deepcopy(current_presets)
+                for _ in range(0, headroom + 1):
```

Comment:
could this be:

```py
test_presets.extend([
   cluster.Structs.PresetStruct(....)] * (headroom+1))
```

readability of a anonymous for loop is odd. why do you need to specify the starting range value of 0 ?